crates/storage/provider/src/transaction.rs

```diff
@@ -719,6 +719,13 @@ where
                 current_transition_id += 1;
             }
 
+            // Transition is incremeneted every time before Paris hardfork and after
+            // Shanghai only if there are Withdrawals in the block. So it is correct to
+            // to increment transition id every time there is a block changeset present.
+            if !results.block_changesets.is_empty() {
+                current_transition_id += 1;
+            }
+
```

Comment:
i'm still thinking about possible repercussions of this. meanwhile, something seems wrong here. current transition id is not selected correctly

Reply:
Wdym about id is not selected correctly?

Check initial issue, we were differently setting transition id, and this worked as every before Paris block had a block reward as changeset.

Between Paris and Shanghai, there isn't any block changesets and transition_id will get wrongly incremented

Reply:
moved order, increment needs to be after insertion.